### PR TITLE
feat(linter) oxc: erasing op

### PR DESCRIPTION
**Pull request by camc314**

### CODE REVIEW

**Comment by camc314** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1834?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #1832** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1832?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1833** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1833?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1834** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1834?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/1834?utm_source=stack-comment).

**Review by Boshen — COMMENTED** [review]
I think we need to polish the help messages a bit.

> This is most likely not the intended outcome.

Probably need some guidance on the action? How does clippy warn about this?

File: crates/oxc_linter/src/snapshots/erasing_op.snap [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+source: crates/oxc_linter/src/tester.rs
+expression: erasing_op
+---
+  ⚠ oxc(erasing-op): Unexpected erasing operation
+   ╭─[erasing_op.tsx:1:1]
+ 1 │ x * 0;
+   · ─────
```
Comment (Boshen) [inline crates/oxc_linter/src/snapshots/erasing_op.snap:8]: We can have the message "This evaluates to 0" here

File: crates/oxc_linter/src/snapshots/erasing_op.snap [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+   ╰────
+  help: This expression will always evaluate to zero. This is most likely not the intended outcome.
```
Comment (Boshen) [inline crates/oxc_linter/src/snapshots/erasing_op.snap:10]: And guidance here.

**Comment by camc314** [comment]
> I think we need to polish the help messages a bit.
> 
> > This is most likely not the intended outcome.
> 
> Probably need some guidance on the action? How does clippy warn about this?

From clippy:

```
warning: this operation will always return zero. This is likely not the intended outcome
  --> crates/oxc_linter/src/rules/oxc/erasing_op.rs:57:17
   |
57 |         let b = a * 0;
   |                 ^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#erasing_op
   = note: `-W clippy::erasing-op` implied by `-W clippy::all`
   = help: to override `-W clippy::all` add `#[allow(clippy::erasing_op)]`
```

I updated the diagnostic to be:

```
  ⚠ oxc(erasing-op): Unexpected erasing operation. This expression will always evaluate to zero.
   ╭─[erasing_op.tsx:1:1]
 1 │ x * 0;
   · ─────
   ╰────
  help: This is most likely not the intended outcome. Consider removing the operation, or directly assigning zero to the variable
  ```

**Review by Boshen — APPROVED** [review]

**Pull request merged**